src/core/iomgr/tcp_client_posix.c

```diff
@@ -83,6 +84,15 @@ static int prepare_socket(const struct sockaddr *addr, int fd) {
             strerror(errno));
     goto error;
   }
+#else
+  if (!grpc_set_socket_nonblocking(fd, 1) || !grpc_set_socket_cloexec(fd, 1) ||
```

Comment:
There's too much duplicated logic in this block.

Can we instead introduce a function: `bool grpc_supports_low_latency(sockaddr *)` that returns true always for the `!GPR_HAVE_UNIX_SOCKET` case, and `addr->sa_family != AF_UNIX` for the `GPR_HAVE_UNIX_SOCKET` case?